### PR TITLE
CAMEL-8799: Make json-path suppress PathNotFoundExceptions by default

### PR DESCRIPTION
**Pull request by pax95**
This PR is a solution for option 1 described in the issue, and the tests can be reused in the implementation of option 2.

### CODE REVIEW

**Comment by davsclaus** [comment]
Thanks for the PR the code has been merged. Do you mind closing this?

**Comment by pax95** [comment]
sure - and thanks !!

File: components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathEngine.java [hunk 1]
```diff
@@ -38,7 +41,8 @@
     private final Configuration configuration;
 
     public JsonPathEngine(String expression) {
-        this.configuration = Configuration.defaultConfiguration();
+        Defaults defaults = DefaultsImpl.INSTANCE;
```
Comment (yihtserns) [inline components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathEngine.java]: This comment is 2 years late, but I don't think there's any need to use json-path's internal API?

`Configuration.Builder.build()` will do the same thing internally anyway if no custom `jsonProvider` provided.

Comment (davsclaus) [reply components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathEngine.java]: @yihtserns you are welcome to provide a PR to fix this

Comment (davsclaus) [reply components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathEngine.java]: Thanks I just pushed a fix for this: https://github.com/apache/camel/commit/9e58df3c602f2502333ca0760e1a0baee1c5a909

Comment (yihtserns) [reply components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathEngine.java]: I wanted to create a PR, but I'm not very sure of the process.

I read that I have to:
1. Create a JIRA ticket
2. Commit and reference that ticket ID in the message
3. Create a PR

Are the steps correct?  And if it is, will that be too much for some refactoring to remove some duplicated lines?

Comment (davsclaus) [reply components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathEngine.java]: If its just some cleanup and minor code changes then you do NOT need to create a JIRA ticket. We only use JIRA for normal/bigger things. So in your case, just do the PR

Comment (yihtserns) [reply components/camel-jsonpath/src/main/java/org/apache/camel/jsonpath/JsonPathEngine.java]: 👌

**Pull request closed without merging**